spec/unit/webrtc/callFeed.spec.ts

```diff
@@ -85,5 +85,129 @@ describe("CallFeed", () => {
                 expect(feed.isVideoMuted()).toBeTruthy();
             });
         });
+
+        describe("voice activity detection", () => {
+            it("voice activity should disable audio track", () => {
+                feed.stream.addTrack(
+                    //@ts-ignore Mock
+                    new MockMediaStreamTrack("track", "audio", true),
+                );
+
+                feed.setVoiceActivityThreshold(Infinity);
+
+                setTimeout(() => {
+                    expect(feed.stream.getAudioTracks()[0].enabled).toBe(false);
+                }, 1000);
+            });
+
+            it("voice activity should enable audio track", () => {
+                feed.stream.addTrack(
+                    //@ts-ignore Mock
+                    new MockMediaStreamTrack("track", "audio", true),
+                );
+
+                feed.setVoiceActivityThreshold(-Infinity);
+                setTimeout(() => {
+                    expect(feed.stream.getAudioTracks()[0].enabled).toBe(true);
+                }, 1000);
+            });
+
+            it("voice activity should enable audio track", () => {
+                feed.stream.addTrack(
+                    //@ts-ignore Mock
+                    new MockMediaStreamTrack("track", "audio", true),
+                );
+
+                feed.setVoiceActivityThreshold(-50);
+                feed.speakingVolumeSamples = [-40];
+
+                setTimeout(() => {
+                    expect(feed.stream.getAudioTracks()[0].enabled).toBe(true);
+                }, 1000);
+            });
+
+            it("voice activity should disable audio track", () => {
+                feed.stream.addTrack(
+                    //@ts-ignore Mock
+                    new MockMediaStreamTrack("track", "audio", true),
+                );
+
+                feed.setVoiceActivityThreshold(-50);
+                feed.speakingVolumeSamples = [-60];
+
+                setTimeout(() => {
+                    expect(feed.stream.getAudioTracks()[0].enabled).toBe(false);
+                }, 1000);
+            });
+
+            it("voice activity should not disable audio track after a few milliseconds", async () => {
+                // Someone speaking
+                // Stops speaking for a few milliseconds
+                // Starts speaking again
+                // -> Is not muted in between (VAD_COOLDOWN)
+
+                feed.stream.addTrack(
+                    //@ts-ignore Mock
```

Comment:
Could we avoid this by adding (or using, if it already exists) a `typed()` method on the mock class that would return the object with correct type cast?

Reply:
To be honest we simply took the existing mute implementation and copied the style from that to ensure we do it in the same manner.
The only thing we did different is that hugos formatting tool kinda broke the style again :D

I will talk to hugo though to see what we can do.

Reply:
Do you mean that we should remove `//@ts-ignore Mock`?
The implementation is the same as in the other tests, so maybe this could be something to be changed in another pull request?